pymove/utils/distances.py

```diff
@@ -51,3 +55,141 @@ def haversine(lat1, lon1, lat2, lon2, to_radians=True, earth_radius=6371):
+    """
+    For each point on a trajectory, it returns the point closest to
+    another trajectory based on the Euclidean distance.
+
+    Parameters
+    ----------
+    traj1: dataframe
+        The input of one trajectory.
+
+    traj2: dataframe
+        The input of another trajectory.
+
+    label_lat: string ("lat" by default)
+        Label of the trajectories dataframe referring to the latitude.
+
+    label_lon: string ("lon" by default)
+        Label of the trajectories dataframe referring to the longitude.
+    """
+
+    result = pd.DataFrame(columns=traj1.columns)
+
+    for i in range(0, len(traj1)):
+        round_result = 999999999999999
+        round_traj = []
+        for j in range(0, len(traj2)):
+            this_distance = distance.euclidean(
+                (traj1[label_lat].iloc[i], traj1[label_lon].iloc[i]),
+                (traj2[label_lat].iloc[j], traj2[label_lon].iloc[j]),
+            )
+            if this_distance < round_result:
+                round_result = this_distance
+                round_traj = traj2.iloc[j]
+        result = result.append(round_traj)
+
+    return result
+
+
+def MEDP(traj1, traj2, label_lat='lat', label_lon='lon'):
+    """
+    Returns the Mean Euclidian Distance Predictive between
+    two trajectories, which considers only the spatial
+    dimension for the similarity measure.
+
+    Parameters
+    ----------
+    traj1: dataframe
+        The input of one trajectory.
+
+    traj2: dataframe
+        The input of another trajectory.
+
+    label_lat: string ("lat" by default)
+        Label of the trajectories dataframe referring to the latitude.
+
+    label_lon: string ("lon" by default)
+        Label of the trajectories dataframe referring to the longitude.
+    """
+
+    soma = 0
+    traj2 = _nearest_points(traj1, traj2, label_lat, label_lon)
+    for i in range(0, len(traj1)):
+        this_distance = distance.euclidean(
+            (traj1[label_lat].iloc[i],
+             traj1[label_lon].iloc[i]),
+            (traj2[label_lat].iloc[i],
+             traj2[label_lon].iloc[i]))
+        soma = soma + this_distance
+    return soma
+
+
+def MEDT(traj1, traj2, label_lat='lat', label_lon='lon', label_time='datetime'):
+    """
+    Returns the Mean Euclidian Distance Trajectory between two
+    trajectories, which considers the spatial dimension and the
+    temporal dimension when measuring similarity.
+
+    Parameters
+    ----------
+    traj1: dataframe
+        The input of one trajectory.
+
+    traj2: dataframe
+        The input of another trajectory.
+
+    label_lat: string ("lat" by default)
+        Label of the trajectories dataframe referring to the latitude.
+
+    label_lon: string ("lon" by default)
+        Label of the trajectories dataframe referring to the longitude.
+
+    label_time: string ("datetime" by default)
+        Label of the trajectories dataframe referring to the timestamp.
+    """
+
+    soma = 0
+    if(len(traj2) > len(traj1)):
```

Comment:
Está sendo repetido o mesmo código nesse `if ... else`.
Acho melhor, como retorna apenas soma, não parece importar quem é `traj1` e `traj2`.
Pode apenas fazer a verificação e um swap.
```python
if(len(traj2) < len(traj1):
    traj1, traj2 = traj2, traj1
```
Parece também que não faz nada se as trajetórias forem do mesmo tamanho. Pode colocar no início também.
```python
soma = 0
if(len(traj2) == len(traj1)):
    return soma
```

Reply:
> Parece também que não faz nada se as trajetórias forem do mesmo tamanho. Pode colocar no início também.
Deveria fazer sim. Esqueci de por um <=